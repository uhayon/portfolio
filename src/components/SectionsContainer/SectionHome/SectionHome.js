import React from 'react';
import PropTypes from 'prop-types';

import styles from './SectionHome.module.scss';

const Title = ({ selectedLanguage }) => (
  <h2>
    {
      selectedLanguage === 'EN' ?
      'Hi! I\'m uRi, a full stack developer' :
      'Hola! Soy uRi, un desarrollador full stack' 
    }
  </h2>
);

const Body = ({ selectedLanguage }) => (
  <>
    {
      selectedLanguage === 'EN' ?
      <p>Programming is my passion. I've been doing it for over 5 years and I'm always looking for ways to grow every day.</p> :
      <p>Programar es mi pasión. He estado haciéndolo por más de 5 años y siempre estoy buscando maneras de crecer y mejorar día a día.</p>
    }
    {
      selectedLanguage === 'EN' ?
      <p>
        I want to deliver products that are way better than expected. 
        Not just aesthetically pleasing; I want them to <span className={styles.important}>add value to your business</span>, helping you <span className={styles.important}>boost performance</span>
      </p> :
      <p>
        Quiero entregar productos que sean mucho mejor de lo esperado.
        No solamente que se vean bien, quiero que <span className={styles.important}>le den valor agregado a tu negocio</span>, ayudándote a <span className={styles.important}>mejorar tu desempeño</span>
      </p>
    }
    {
      selectedLanguage === 'EN' ?
      <h4>If there is anything on the site that is of interest to you or you have any questions at all, please do not hesitate to reach out!</h4> :
      <h4>Si viste algo que fue de tu interés o tenés alguna consulta, por favor no dudes en contactarme!</h4>
    }
  </>
)

const SectionHome = ({ selectedLanguage }) => (
    <section id='home' className={styles.sectionHome}>
      <div className={styles.bodyContainer}>
        <div className={styles.textsContainer}>
          <Title selectedLanguage={selectedLanguage} />
          <Body selectedLanguage={selectedLanguage} />
          <button className={styles.contactButton} onClick={() => window.location.href = '#contact'}>
            {
              selectedLanguage === 'EN' ?
              'Contact Me' :
              'Contactame'
            }
          </button>
        </div>
      </div>
    </section>
);

SectionHome.propTypes = {
  selectedLanguage: PropTypes.string.isRequired
}

export default SectionHome;