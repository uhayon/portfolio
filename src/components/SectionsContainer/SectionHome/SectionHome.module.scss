@import '../../../globalStyles/colors.scss';

.sectionHome {
    background: url('../../../assets/bg_portfolio.jpeg') no-repeat center center fixed;
    background-size: cover;
    height: 100vh;
    width: 100vw;
    display: flex;
    justify-content: center;
    align-items: center;

    .bodyContainer {
        height: 80%;
        width: 100%;
        @media screen and (min-width: 500px) {
            width: 75%;
        }

        .textsContainer {
            background-color: rgba(0, 0, 0, .75);
            color: $white;
            text-align: center;
            padding: 2rem;

            .contactButton {
                background-color: $red;
                color: $white;
                font-size: 1.5rem;
                font-weight: bold;
                padding: 1rem 1.5rem;
                border: none;
                border-radius: 100px;
                transform: scale(1);
                transition: all 0.2s linear;
                margin: 2rem 0;
                cursor: pointer;
                outline: none;

                &:hover {
                    transform: scale(1.2);
                    background-color: $red-light
                }
            }

            .important {
                font-weight: bold;
                font-style: italic;
                color: $red;
            }
        }
    }
}