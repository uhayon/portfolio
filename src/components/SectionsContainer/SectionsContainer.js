import React from 'react';

const SectionsContainer = () => (
  <>
  </>
);

export default SectionsContainer;