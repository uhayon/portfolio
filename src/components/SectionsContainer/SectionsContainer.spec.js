import React from 'react';
import { shallow } from 'enzyme';

import SectionsContainer from './SectionsContainer';
import SectionHome from './SectionHome/SectionHome';
// import SectionSkills from './SectionSkills/SectionSkills';
// import SectionProjects from './SectionProjects/SectionProjects';
// import SectionContact from './SectionContact/SectionContact';

describe('Mouting Sections Container', () => {
  let wrapper;
  beforeEach(() => wrapper = shallow(<SectionsContainer />));

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('Should have the SectionHome, SectionSkills, SectionProject and SectionContact as children', () => {
    const childrenSections = [SectionHome/*, SectionSkills, SectionProjects, SectionContact*/];
    // expect(wrapper.prop('children').length).toBeDefined();
    // expect(wrapper.prop('children').length).toEqual(4);
    childrenSections.forEach(child => {
      expect(wrapper.exists(child));
    })
  })
});