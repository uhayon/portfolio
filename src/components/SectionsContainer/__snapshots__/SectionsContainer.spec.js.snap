// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mouting Sections Container Should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SectionsContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <SectionHome />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "selectedLanguage": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <SectionHome />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "selectedLanguage": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
